Sources/MediaCategories/MediaCategoryViewController.swift

```diff
@@ -319,6 +319,16 @@ extension VLCMediaCategoryViewController: VLCEditControllerDelegate {
     func editController(editController: VLCEditController, cellforItemAt indexPath: IndexPath) -> MediaEditCell? {
         return collectionView.cellForItem(at: indexPath) as? MediaEditCell
     }
+
+    func editController(editController: VLCEditController,
+                        needToPresent viewController: UIViewController) {
+        // Create a new navigation controller here in order to have a separate hierachy inside
+        //  the "add to playlist" logic. Indeed, having a separate navigationController allows
+        //  having a separate hierarchy for playlists, for example having folders for playlists.
+
+        let newNavigationController = UINavigationController(rootViewController: viewController)
+        navigationController?.present(newNavigationController, animated: true, completion: nil)
+    }
```

Comment:
Thoughts? I could also cache the navigationController.

Reply:
looks good and you shouldn't cache it. You don't need the comment and I would name the method simply ```present```  instead of ```needToPresent```